jina/peapods/peas/factory.py

```diff
@@ -25,6 +31,10 @@ def build_pea(
         """
         # copy to update but forward original
         cargs = deepcopy(args)
+        if cargs.host != __default_host__ and not cargs.disable_remote:
+            cargs.timeout_ready = -1
```

Comment:
set here `cargs.disable_remote` so that we make sure that this does not enter an infinite loop